lib/MauticApi.php

```diff
@@ -53,15 +54,11 @@ public static function getContext($apiContext, AuthInterface $auth, $baseUrl = '
     /**
      * Get an API context object.
      *
-     * @param string        $apiContext API context (leads, forms, etc)
-     * @param AuthInterface $auth       API Auth object
-     * @param string        $baseUrl    Base URL for API endpoints
-     *
      * @return Api\Api
      *
      * @throws ContextNotFoundException
      */
-    public function newApi($apiContext, AuthInterface $auth, $baseUrl = '')
+    public function newApi(string $apiContext, AbstractAuth $auth, string $baseUrl = '')
```

Comment:
Changing a public or protected method signature is a BC change as someone can extend the method in their code and so it would cause errors with this change. Why is the change from AuthInterface to AbstractAuth needed? I'd prefer the interface as it's more SOLID. 

Similar change was done in the other file too.

Reply:
@dennisameling ⬆️ ❓